src/lib.rs

```diff
@@ -114,97 +114,3630 @@ use std::{
     ffi::OsString,
     fmt::{self, Display, Formatter},
     io::{Error, ErrorKind},
+    env,                    // for environment variables (LANG, LC_ALL)
 };
 
+/// output : en_US.UTF-8
+/// en     : language
+/// US     : country
+#[allow(dead_code)]
+fn get_language_and_country() -> String {
+    if let Ok(language) = env::var("LC_ALL") {
+        // possible output : en_US.UTF-8
+        return language;
+    } else if let Ok(language) = env::var("LANG") {
+        // possible output : en_US.UTF-8
+        return language;
+    } else if let Ok(language) = env::var("LANGUAGE") {
+        // possible output : en_US.UTF-8
+        return language;
+    }
+    String::new()
+}
+
+
 /// This crate's convenience type alias for [`Result`](std::result::Result)s
 pub type Result<T = (), E = Error> = std::result::Result<T, E>;
 
-/// Region code for a [`Language`] dialect
-///
-/// Uses <https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2>
-#[non_exhaustive]
-#[repr(u32)]
-#[derive(Copy, Clone, Eq, PartialEq, Debug)]
-pub enum Region {
-    // FIXME: V2: u32::from_ne_bytes for region codes, with `\0` for unused
-    // FIXME: Add aliases up to 3-4 letters, but hidden
-    /// Any dialect
-    Any,
-    /// `US`: United States of America
+/// Region code for a [`Language`] dialect
+///
+/// Uses <https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2>
+///
+/// https://www.suny.edu/media/suny/content-assets/documents/international-student/InternationalCountryCodes.pdf
+/// https://www.iban.com/country-codes
+#[non_exhaustive]
+#[repr(u32)]
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub enum Region {
+    // FIXME: V2: u32::from_ne_bytes for region codes, with `\0` for unused
+    // FIXME: Add aliases up to 3-4 letters, but hidden
+    /// Any dialect
+    Any,
+
+    /// an u32 code for region
+    #[doc(hidden)]
+    Custom(u32),
```

Comment:
I think we'll probably want to remove variant this before merging, and move the numeric parsing in a [`FromStr`](https://doc.rust-lang.org/std/str/trait.FromStr.html) implementation.